plugins/services/src/js/reducers/serviceForm/Volumes.js

```diff
@@ -0,0 +1,107 @@
+import {
+  ADD_ITEM,
+  REMOVE_ITEM,
+  SET
+} from '../../../../../../src/js/constants/TransactionTypes';
+
+module.exports = {
+  JSONReducer(state, {type, path, value}) {
+    if (path == null) {
+      return state;
+    }
+
+    // `this` is a context which is given to every reducer so it could
+    // cache information.
+    // In this case we are caching an two array's one for the localVolumes
+    // and one for externalVolumes we need this so that there index is
+    // fitting with the ones in ExternalVolumes. we combine them before
+    // returning.
+    if (this.externalVolumes == null) {
+      this.externalVolumes = [];
+    }
+
+    if (this.localVolumes == null) {
+      this.localVolumes = [];
+    }
+
+    const joinedPath = path.join('.');
+
+    if (joinedPath.search('externalVolumes') !== -1) {
+      if (joinedPath === 'externalVolumes') {
+        switch (type) {
+          case ADD_ITEM:
+            this.externalVolumes.push({
+              containerPath: null, external: {
+                name: null,
+                provider: 'dvdi',
+                options: {
+                  'dvdi/driver': 'rexray'
+                }
+              },
+              mode: 'RW'
+            });
+            break;
+          case REMOVE_ITEM:
+            this.externalVolumes =
+              this.externalVolumes.filter((item, index) => {
+                return index !== value;
+              });
+            break;
+        }
+
+        return [].concat(this.externalVolumes, this.localVolumes);
+      }
+
+      const index = joinedPath.match(/\d+/)[0];
```

Comment:
I don't really understand why you need a regex to get the numerical component of a path. I would still vote for `path[1]` since you always know at which index you will find the path.